### ISSUE TITLE
RDS Deletion Protection API Support (aws_rds_cluster)

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#6007]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Go SDK v1.15.43 (dependency PR #6002) includes a new `DeletionProtection` boolean parameter:

```
	// Indicates if the DB cluster should have deletion protection enabled. The
 	// database can't be deleted when this value is set to true. The default is
 	// false.
 	DeletionProtection *bool `type:"boolean"`
```

It is implemented in the following API types:

* `CreateDBClusterInput`
* `ModifyDBClusterInput`
* `RestoreDBClusterFromSnapshotInput`
* `RestoreDBClusterFromS3Input`

### New or Affected Resource(s)

* `aws_rds_cluster`

### Potential Terraform Configuration

```hcl
# Example implementation, details may change during development
resource "aws_rds_cluster" "example" {
  # ... other configuration ...

  deletion_protection = true
}
```

### References

* #6002 
* #6006 (`aws_db_instance` support)

**Comment by bflad** [comment]
Pull request submitted: #6010

**Comment by ozbillwang** [comment]
Nice job, I am waiting for the PR to be merged. 

But @bflad, I have several questions:

1. Should we remove the behavior  `prevent_destroy` out of the resource `aws_rds_cluster` or we should go with the same place to change the action `prevent_destroy` as the new feature of `Deletion Protection`, more than keep both of them with confusion?

What if someone set `prevent_destroy` to true and `deletion_protection` to false, what will happen?

**Comment by bflad** [comment]
Support for this has been merged and will release with version 1.39.0 of the AWS provider, likely later today.
